### DISCUSSION TITLE
How to start with tts wavernn

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#206] [Archived]
<USER>>>> Jalau<br/> <DATE> [May 20, 2020, 11:57pm]<br/> <br/> I am new to the world of deep learning and all that stuff so forgive me<br/> for not knowing anything about it. But I am happy to learn. *slash* <br/> So I have seen the model Tacotron2-iter-260K with a soundcloud link that<br/> sounds awesome. However having successfully deployed it after a lot of<br/> trouble shooting ended up being not as fulfilling as I expected it to<br/> be. It sounded much worse. Now after digging deeper I have noticed that<br/> it probably was used in combination with WaveRNN. But how do I continue<br/> from here? I setup WaveRNN with the pre trained modules from here: *slash* <br/> <br/> ![](https://github.githubassets.com/favicons/favicon.svg){.site-icon<br/> [GitHub](https://github.com/erogol/WaveRNN#released-models)<br/> <br/> ### [erogol/WaveRNN](https://github.com/erogol/WaveRNN#released-models)<br/> <br/> Pytorch implementation of Deepmind *slash* 's WaveRNN model - erogol/WaveRNN<br/> <br/> *slash* <br/> How do I use it in combination with the tacotron2 model? How do I get<br/> the expected results as in the soundcloud links for the tacotron2<br/> model? *slash* <br/> And is there anyway to speed up the process of WaveRNN? I have a 2080<br/> Super and a 3900x but both are chilling at 10% usage and WaveRNN takes<br/> like 2 minutes for one short sentence. *slash* <br/> I am intending to use this for a real time text to speech application so<br/> I would be happy if there was a way to achieve speeds around like 5<br/> seconds max. Thanks for any help in advance.<br/> <br/> Best regards *slash* <br/> Jalau<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/how-to-start-with-tts-wavernn]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [May 21, 2020, 1:29am]<br/> <br/> Hi [ *slash* Jalau](<br/> to the forum! Not sure which resources you've looked at but I'd suggest<br/> the models page (and ideally reading around the various bits of the wiki<br/> to pick up what you can)<br/> <br/> ![](https://github.githubassets.com/favicons/favicon.svg){.site-icon<br/> [GitHub](https://github.com/mozilla/TTS/wiki/Released-Models)<br/> <br/> ### [mozilla/TTS](https://github.com/mozilla/TTS/wiki/Released-Models)<br/> <br/> forum: https://discourse.mozilla.org/c/tts) - mozilla/TTS<br/> <br/> The output will be improved when using a vocoder (ie WaveRNN but also<br/> PWGAN or MelGAN, these last two being faster for inference).<br/> <br/> You've got a few options, including installing the self contained<br/> package at the bottom of the model page, or installing the particular<br/> models based on the relevant commits.<br/> <br/> Another thing that might be handy if you're starting out and want to see<br/> how things fit together is to look at some of the Colabs that people<br/> have linked to. There's a couple of fairly recent ones on this page<br/> here: <https://github.com/mozilla/TTS/issues/345><br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> Jalau<br/> <DATE> [May 21, 2020, 11:39am]<br/> <br/> Thanks for the quick answer<br/> ! I haven't fully read<br/> through the wiki yet so I will do that. However I already took a look at<br/> the model section. *slash* <br/> I have already installed this model:<br/> <https://github.com/mozilla/TTS/tree/Tacotron2-iter-260K-824c091> *slash* <br/> It works but the result is obviously not the same as in the example<br/> soundcloud link on the models page. I have also tried the pre packaged<br/> pip install for a different model and I got it running too but it<br/> sounded even worse than the Tacotron2 260k model. *slash* <br/> So basically after a lot of troubleshooting I was able to start the demo<br/> server with the corresponding models howevever I don't know how a<br/> vocoder can be added on top of that to improve the quality. *slash* <br/> So basically how do I combine a vocoder with an existing TTS demo server<br/> to get a combined improved output from both? *slash* <br/> Thanks in advance!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [May 21, 2020, 3:25pm]<br/> <br/> High level: you can simply install the repo for the relevent vocoder and<br/> then adjust the configuration for the TTS demo server.<br/> <br/> To actually do that, I was suggesting having a look at the Colabs as<br/> they literally do all the steps needed to get them to work together, so<br/> you'd see how they fit and could then mimic that locally.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> Jalau<br/> <DATE> [May 21, 2020, 4:09pm]<br/> <br/> Thanks! I will definitely<br/> take a look at those collab repositories. Is there a world where one<br/> could get those good sounding results in a matter of seconds or will it<br/> always take like a minute to render one sentence?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> nmstoker<br/> <DATE> [May 21, 2020, 4:55pm]<br/> <br/> I don't have PC / GPU figures to hand. I've mostly used MelGAN and it's<br/> reasonably fast (ie in the order of seconds). I *think* PWGAN is quick<br/> too and then WaveRNN is slower but generally the highest quality.<br/> <br/> [<br/> [[Jerson_Luiz_de_Paula_Junior<br/> <DATE> [May 21, 2020, 8:40pm]<br/> <br/> Hello, I am also learning and use this example below that can be used in<br/> google colab with GPU:<br/> <br/> <https://colab.research.google.com/github/tugstugi/dl-colab-notebooks/blob/master/notebooks/Mozilla_TTS_WaveRNN.ipynb#scrollTo=klsVLR6w_u4P><br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [May 22, 2020, 5:51pm]<br/> <br/> Thanks for the colab. I wanted to try a different vocoder (the universal<br/> ), but it is made<br/> for 16 KHz instead of my trained Tacotron with 22.5 KHz.<br/> <br/> I am not the signals expert, any idea on how to downsample between the<br/> two or is there some other thing I'm missing here? Because now I have<br/> Donald Duck talking<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 22, 2020, 6:06pm]<br/> <br/> You have two options. Either finetune WaveRNN with 22050 features (I<br/> have tried it and there is a lot of noise), or finetune Taco2 with 16khz<br/> audio, by restoring your TTS model.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [May 22, 2020, 8:47pm]<br/> <br/> Thanks, I thought there might be an easy way ... how foolish of me<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [May 22, 2020, 8:54pm]<br/> <br/> I find vocoders in general to be tricky<br/> I spent a lot of time trying to train WaveRNN but I didn't really get it<br/> work, so now I am trying to make PWGAN work. The original PWGAN repo has<br/> some super nice universal vocoders which you can try, if you trained<br/> your TTS model with mean-var normalization.<br/> <br/> [Archived Post]<br/>
